compiler/rustc_builtin_macros/src/test.rs

```diff
@@ -107,6 +107,36 @@ pub fn expand_test_or_bench(
         return vec![];
     }
 
+    let not_testable_error = |item: Option<&ast::Item>| {
```

Comment:
I think this might be a bit cleaner if this was a function rather than a closure.